### PR TITLE
Show web app state in explorer

### LINKED ISSUE #78: Show the service's state
We allow stopping, starting, etc. But we can't tell if the service is running or stopped right now. Add a state indicator.

image

### PR DESCRIPTION
**Pull request by whiskeyjay**
Fix #78

### CODE REVIEW

File: src/explorer/SiteTreeItem.ts [hunk 1]
```diff
@@ -110,6 +120,12 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
     public async editScmType(client: WebSiteManagementClient): Promise<string> {
         return await this.siteWrapper.editScmType(client);
     }
+
+    private async reloadSite(client: WebSiteManagementClient): Promise<void> {
```
Comment (nturinski) [inline src/explorer/SiteTreeItem.ts]: I think the start/stop functions use the Azure API to confirm that the web app has started/stopped.  Do we need to reacquire the resource through the API to update (Running)/(Stopped) and reconstruct the SiteWrapper to confirm the state?

File: src/explorer/DeploymentSlotTreeItem.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export class DeploymentSlotTreeItem extends SiteTreeItem {
     }
 
     public get label(): string {
-        return this.siteWrapper.slotName;
+        return `${this.siteWrapper.slotName} (${this.site.state})`;
```
Comment (nturinski) [inline src/explorer/DeploymentSlotTreeItem.ts]: nit: Having all the web apps say (Running) makes the explorer look very cluttered.  What do you think about only displaying (Stopped) and omitting (Running) for apps that are on?

File: src/explorer/SiteTreeItem.ts (hunk 1 above)
Comment (whiskeyjay) [reply src/explorer/SiteTreeItem.ts]: Good point! I'll make a bit of adjustment...

File: src/explorer/SiteTreeItem.ts [hunk 3]
```diff
@@ -49,6 +49,16 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
...
+        await this.siteWrapper.start(client);
+        await this.reloadSite(client);
+    }
+
+    public async stop(client: WebSiteManagementClient): Promise<void> {
+        await this.siteWrapper.stop(client);
+        await this.reloadSite(client);
+    }
+
     public browse(): void {
```
Comment (nturinski) [inline src/explorer/SiteTreeItem.ts]: Currently if a web app is Stopped and then you "Restart" the app through the context menu, the state doesn't refresh unless you manually refresh the node.

Comment (whiskeyjay) [reply src/explorer/SiteTreeItem.ts]: Yep, I'll handle that case.

File: src/explorer/SiteTreeItem.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply src/explorer/SiteTreeItem.ts]: No need to reload the site or reconstruct the sitewrapper. Just use the existing `siteWrapper.getState` method

File: src/extension.ts [hunk 4]
```diff
@@ -83,7 +83,8 @@ export function activate(context: vscode.ExtensionContext): void {
         const siteType = util.isSiteDeploymentSlot(node.treeItem.site) ? 'Deployment Slot' : 'Web App';
         outputChannel.show();
         outputChannel.appendLine(`Starting ${siteType} "${node.treeItem.site.name}"...`);
-        await node.treeItem.siteWrapper.start(nodeUtils.getWebSiteClient(node));
+        await node.treeItem.start(nodeUtils.getWebSiteClient(node));
+        node.treeDataProvider.refresh(node, false);
```
Comment (EricJizbaMSFT) [inline src/extension.ts]: I thought this didn't work in the released version of VS Code - don't you still need to refresh the parent?

In any case, I would prefer if you used `node.refresh()` and then we only have to change it in one place (aka [here](https://github.com/Microsoft/vscode-azuretools/blob/master/ui/src/treeDataProvider/AzureNode.ts#L62)) once VS Code releases the tree fix

**Pull request merged**